pydarn/io/borealis.py

```diff
@@ -0,0 +1,1298 @@
+                    # (only used for last ranges)
+
+                    # (num_ranges x num_lags, flattened)
+                    flattened_data = np.array(this_correlation).flatten()
+
+                    int_data = np.empty(
+                        flattened_data.size * 2, dtype=np.float32)
+                    int_data[0::2] = flattened_data.real
+                    int_data[1::2] = flattened_data.imag
+                    # num_ranges x num_lags x 2
+                    new_data = int_data.reshape(
+                        v['correlation_dimensions'][1], 
+                        v['correlation_dimensions'][2]-1, 2)
+                    # NOTE: Flattening happening in 
+                    # convert_to_dmap_datastructures
+                    # place the darn-style array in the dict
+                    correlation_dict[key] = new_data
+
+                record_dict = {
+                    'radar.revision.major': np.int8(borealis_major_revision),
+                    'radar.revision.minor': np.int8(borealis_minor_revision),
+                    'origin.code': np.int8(100),  # indicating Borealis origin
+                    'origin.time': datetime.utcfromtimestamp(
+                        v['sqn_timestamps'][0]).strftime("%c"),
+                    'origin.command': 'Borealis ' + v['borealis_git_hash'] + \
+                        ' ' + v['experiment_name'],
+                    'cp': np.int16(v['experiment_id']),
+                    'stid': np.int16(code_to_stid[v['station']]),
+                    'time.yr': np.int16(datetime.utcfromtimestamp(
+                        v['sqn_timestamps'][0]).year),
+                    'time.mo': np.int16(datetime.utcfromtimestamp(
+                        v['sqn_timestamps'][0]).month),
+                    'time.dy': np.int16(datetime.utcfromtimestamp(
+                        v['sqn_timestamps'][0]).day),
+                    'time.hr': np.int16(datetime.utcfromtimestamp(
+                        v['sqn_timestamps'][0]).hour),
+                    'time.mt': np.int16(datetime.utcfromtimestamp(
+                        v['sqn_timestamps'][0]).minute),
+                    'time.sc': np.int16(datetime.utcfromtimestamp(
+                        v['sqn_timestamps'][0]).second),
+                    'time.us': np.int32(datetime.utcfromtimestamp(
+                        v['sqn_timestamps'][0]).microsecond),
+                    'txpow': np.int16(-1),
+                    'nave': np.int16(v['num_sequences']),
+                    'atten': np.int16(0),
+                    'lagfr': np.int16(v['first_range_rtt']),
+                    'smsep': np.int16(1e6/v['rx_sample_rate']),
+                    'ercod': np.int16(0),
+                    'stat.agc': np.int16(0),
+                    'stat.lopwr': np.int16(0),
+                    'noise.search': np.float32(v['noise_at_freq'][0]),
+                    'noise.mean': np.float32(0),
```

Comment:
I cannot remember, but why can't we calculate `noise.mean` right now?

Reply:
requires clear freq search or a listening period that is not currently built in